### ISSUE TITLE
Item progression works for every upgrade when EnableRandomizeUpgrades is active

### ISSUE THREAD

**Issue by tixomirof** [Malcolm-Q/LC-LateGameUpgrades#580]
With `Enable Randomize Upgrades = true`, it does not matter which upgrades you have in lategame store list, if you sell items for item progression, it will apply to all upgrades in the mod.

Other config options from randomize upgrades category:
```
[_Randomize Upgrades_]

## Upgrades will be randomized per selected event to be allowed purchased on the store.
# Setting type: Boolean
# Default value: false
Enable Randomize Upgrades = true

## Amount of Upgrades that will always appear in the store (if allowed).
# Setting type: Int32
# Default value: 5
Amount of Upgrades = 10

## Wether purchased upgrades will always show in the store and the specified amount of random upgrades will not include these in the selection.
# Setting type: Boolean
# Default value: true
Show purchased upgrades = false

## Event which triggers the random upgrades being changed in the store.
# Setting type: RandomizeUpgradeEvents
# Default value: PerQuota
# Acceptable values: PerQuota, PerMoonRouting, PerMoonLanding
Randomize Upgrades Event = PerQuota
```

_Ex: you have only running shoes in lategame store. Tattered metal sheet contributes to running shoes and back muscles. Upon selling tattered metal sheet, it will give points both to running shoes and back muscles. But expected to give only for running shoes._

**Comment by WhiteSpike** [comment]
So you wish item progression to not affect upgrades hidden from randomize?
The two systems were made without considering each other thus why you are describing this.

**Comment by tixomirof** [comment]
> So you wish item progression to not affect upgrades hidden from randomize?
Yeah, exactly, if that's not a problem, of course

**Comment by WhiteSpike** [comment]
3.11.0 release should address this. If you can check, I would appreciate.

**Comment by tixomirof** [comment]
Still present in 3.11.0
What did I do to test it out:
1. Installed Imperium and LGU
2. Configured LGU for item progression route mode and enabled randomize upgrade, 5 per quota
3. Created a new save
4. Went to the company building
5. Gave myself scrap via Imperium to sell them and reach the quota. I had used items that are listed for every other upgrade
6. Took off the planet, recieved the next quota
7. Checked LGU store again - all previously hidden upgrades had recieved progress too.
Then I closed the game, removed randomize upgrades and joined the save again. Screenshot below clearly shows that all upgrades had progressed.

![15-12-2024_234042132](https://github.com/user-attachments/assets/20e8913e-1478-41f4-9e52-bf162bc25033)

What is also noticeable, is that some of the upgrades had negative pricing. Upgrades which should have been upgraded upon progress completition, recieved negative price instead. However, upgrades that were shown among those 5 randomized upgrades, recieved the next upgrade level safely (if they reached 0 or below progress).

**Comment by WhiteSpike** [comment]
So you want them to do nothing when the related upgrade is hidden, rather than keep the contribution going but just not applying the upgrade til it gets unhidden again?

**Comment by tixomirof** [comment]
Not really, I want the upgrade to still apply it's functions while it's hidden, but I don't want sold items to contribute to progress for currently hidden upgrades

**Example:**
_I have Hand of Midas Lv0 in the LGU list. I also have Running Shoes Lv0 which is not shown in the LGU store but is enabled in the config. Let's say, they both require Clown Horn for progression. Whenever I sell Clown Horn worth 60, it will change prices of Hand of Midas 1000 -> 940 (assuming sell item progress coefficient is 100%) and Running Shoes 600 -> 600._

**Example 2:**
_I have Hand of Midas Lv0 in the LGU list. I also have Running Shoes Lv1 which is not shown in the LGU store due to config key 'ShowPurchasedUpgrades' set to false. Running shoes still functions and gives bonus movement speed. However upon Clown Horn sell, it's price remains the same, unlike hand of midas'._

**Example 3:**
_I have Hand of Midas Lv0 in the LGU list. I also have Running Shoes Lv0 which previously had progress with other items but is currently not shown in the LGU store due to quota rotation, and has the progress of, let's say, 325 (from base 600). I sell Clown Horn worth 60: Hand of Midas' price 1000 -> 940, Running Shoes' price 325 -> 325._

I'm sorry for late responses

**Comment by WhiteSpike** [comment]
Yeah okay so you only want the contribution to not execute when its hidden. That was what I tried describing in the earlier comment, lol.

**Comment by tixomirof** [comment]
> but just not applying the upgrade til it gets unhidden again

I thought you meant the upgrade would disable completely until its shown again 😅 My bad
